suse_migration_services/prechecks/kernels.py

```diff
@@ -54,24 +69,63 @@ def multiversion_and_multiple_kernels():
                         'Please ensure it is set as:\n'
                         "'multiversion.kernels = latest,running'")
         elif set(kernel_multi_version_settings.split(',')) != {'running', 'latest'}:
-            log.warning('The config option multiversion.kernels is not set '
-                        'correctly in /etc/zypp/zypp.conf. It is currrently '
-                        'set as \n%s.\nPlease ensure it is set as:\n'
-                        "'multiversion.kernels = latest,running'",
-                        kernel_multi_version_settings)
+            if fix:
+                log.info("The '--fix' option was provided, "
+                         "setting 'multiversion.kernels = latest,running' ")
+                existing = 'multiversion.kernels = ' + kernel_multi_version_settings
+                correction = 'multiversion.kernels = latest,running'
+                sed_arg = 's/' + existing + '/' + correction + '/'
+                try:
+                    Command.run(["sed", "-i", sed_arg, Defaults.get_zypp_config_path()])
+                except DistMigrationCommandException:
+                    log.error("ERROR: Unable to update /etc/zypp/zypp.conf")
+            else:
+                log.warning("The config option multiversion.kernels is not set "
+                            "correctly in /etc/zypp/zypp.conf. It is currrently "
+                            "set as:\n'multiversion.kernels = %s'.\n"
+                            "Please ensure it is set as:\n"
+                            "'multiversion.kernels = latest,running'",
+                            kernel_multi_version_settings)
+
+
+def multiple_kernels_installed(fix, target, log):
+    """Check for multiple kernels installed"""
 
     kernel_type = 'kernel-default'
-    running_kernel = Command.run(["uname", "-r"])
+    running_kernel_path = '/' + Defaults.get_target_kernel()
+    rpm_command = ["rpm", "-qa", kernel_type]
+    if target:
+        running_kernel_path = Defaults.get_system_root_path() + running_kernel_path
+        rpm_command = \
+            [
+                "chroot",
+                Defaults.get_system_root_path(),
+                "rpm",
+                "-qa",
+                kernel_type
+            ]
+
+    running_kernel = os.readlink(running_kernel_path)
```

Comment:
can we move this statement inside the `if multiple_kernels` ?

Reply:
actually it is in the wrong place. but it should not be in the 'if fix' since kernel_type is used in the rpm_command variable near the top of the function. It should be moved up. Thanks for having me review this and catching the error. I need to see why my test did not catch this error.